source/common/upstream/upstream_impl.cc

```diff
@@ -729,14 +730,16 @@ bool BaseDynamicClusterImpl::updateDynamicHostList(const HostVector& new_hosts,
       hosts_added.push_back(host);
 
       // If we are depending on a health checker, we initialize to unhealthy.
-      if (depend_on_hc) {
+      if (health_checker_ != nullptr) {
         hosts_added.back()->healthFlagSet(Host::HealthFlag::FAILED_ACTIVE_HC);
       }
     }
   }
 
+  const bool skip_healthy_hosts =
```

Comment:
nit: s/skip_healthy_hosts/dont_remove_healthy_hosts (or something like it) I think would be a bit more clear.